src/backend/utils/misc/timeout.c

```diff
@@ -275,9 +275,11 @@ handle_sig_alarm(SIGNAL_ARGS)
 
 	/*
 	 * SIGALRM is always cause for waking anything waiting on the process
-	 * latch.
+	 * latch.  Cope with MyLatch not being there, as the startup process also
+	 * uses this signal handler.
 	 */
-	SetLatch(MyLatch);
+	if (MyLatch)
+		SetLatch(MyLatch);
```

Comment:
According to the description in this [commit](https://github.com/arenadata/gpdb/commit/59f71a0d0b56b2df48db4bf1738aece5551f7a47), the latch becomes usable after `InitPostmasterChild()`. It seems that this function is executed in [do_start_bgworker](https://github.com/arenadata/gpdb/blob/02ab15ccb5dceab6cba070e350d8696cba944a4e/src/backend/postmaster/postmaster.c#L6455). Can we postpone the timer initialization? Is it correct that we don't set the latch in the `SIGALRM` signal handler? The latch is used so that the process can detect that a signal has arrived and resume execution.

Reply:
According to this commit the latch is initialized for *children* of postmaster, not postmaster itself. Without this patch the postmaster would have a segfault here (I checked the coredump, `MyLatch == NULL` here). This means that any timer activations in postmaster would have segfaulted before this fix. Not setting the latch here is acceptable because the latch is not used anywhere else in postmaster.